lib/engine/game/g_1849/game.rb

```diff
@@ -771,12 +824,137 @@ def bank_sort(corporations)
+          @log << "-- Event: #{EVENTS_TEXT[:buy_tokens][1]} --"
+        end
+
+        def event_bonds!
+          @log << "-- Event: #{EVENTS_TEXT[:bonds][1]} --"
+        end
+
+        def event_e_tokens!
+          @log << "-- Event: #{EVENTS_TEXT[:e_tokens][1]} --"
+        end
+
+        # code below is for Bonds variant
+        NUM_LOANS = 6
+        MAXIMUM_LOANS = 1
+        LOAN_VALUE = 500
+        INTEREST_RATE = 50
+
+        def init_loans
+          Array.new(num_loans) { |id| Loan.new(id, loan_value) }
+        end
+
+        def num_loans
+          bonds? ? NUM_LOANS : 0
+        end
+
+        def maximum_loans(_entity)
+          MAXIMUM_LOANS
+        end
+
+        def loan_value(_entity = nil)
+          LOAN_VALUE
+        end
+
+        def interest_rate
+          INTEREST_RATE
+        end
+
+        def interest_owed_for_loans(loans)
+          interest_rate * loans
+        end
+
+        def interest_owed(entity)
+          interest_owed_for_loans(entity.loans.size)
+        end
+
+        def loans_due_interest(entity)
+          entity.loans.size
+        end
+
+        def log_interest_payment(entity, amount)
+          amount_fmt = format_currency(amount)
+          @log << "#{entity.name} pays #{amount_fmt} interest for its issued bond"
+        end
+
+        def take_loan(entity, loan)
+          raise GameError, 'Cannot issue bond' unless can_take_loan?(entity)
+
+          name = entity.name
+          @log << "#{name} issues its bond and receives #{format_currency(loan_value)}"
+          @bank.spend(loan_value, entity)
+          entity.loans << loan
+          @loans.delete(loan)
+          @round.issued_bond[entity] = true
+
+          initial_sp = entity.share_price.price
+          @stock_market.move_left(entity)
+          @log << "#{entity.name}'s share price changes from" \
+                  " #{format_currency(initial_sp)} to #{format_currency(entity.share_price.price)}"
+        end
+
+        def can_take_loan?(entity)
+          bonds? &&
+            entity.corporation? &&
+            !@round.repaid_bond[entity] &&
+            entity.loans.size < maximum_loans(entity) &&
+            !@phase.status.include?('no_bonds')
+        end
+
+        def can_pay_interest?(entity, extra_cash = 0)
+          entity.cash + extra_cash >= interest_owed(entity)
+        end
+
+        # code below is for the Electric Dreams variant
+
+        def status_array(corporation)
+          return unless corporation.e_token
+
+          ['E-Token Purchased']
+        end
+
+        def set_e_token_sold
+          @e_token_sold = false
+        end
+
+        def init_train_handler
+          trains = game_trains.flat_map do |train|
+            Array.new((train[:num] || num_trains(train))) do |index|
+              Train.new(**train, index: index)
+            end
+          end
+
+          electric_dreams? ? G1849::Depot.new(trains, self) : self.class::DEPOT_CLASS.new(trains, self)
```

Comment:
All the logic seems to be handled in 1849 depot for both cases, so why check here? Instead use the 49 depot always?